src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CompletionSource.cs

```diff
@@ -234,15 +226,132 @@ static void CheckForExperimentStatus(ITextView textView, in CompletionOptions op
 
             var document = triggerLocation.Snapshot.GetOpenDocumentInCurrentContextWithChanges();
             if (document == null)
+                return VSCompletionContext.Empty;
+
+            // The computation of completion items is divided into two tasks:
+            //
+            // 1. "Core" items (i.e. non-expanded) which should be included in the list regardless of the selection of expander.
+            //    Right now this includes all items except those from unimported namespaces.
+            //    
+            // 2. Expanded items which only show in the completion list when expander is selected, or by default if the corresponding
+            //    features are enabled. Right now only items from unimported namespaces are associated with expander. 
+            //
+            // #1 is the essence of completion so we'd always wait until its task is completed and return the results. However, because we have
+            // a really tight perf budget in completion, and computing those items in #2 could be expensive especially in a large solution
+            // (e.g. requires syntax/symbol indices and/or runs in OOP,) we decide to kick off the computation in parallel when completion is
+            // triggered, but only include its results if it's completed by the time task #1 is completed, otherwise we don't wait on it and
+            // just return items from #1 immediately. Task #2 will still be running in the background (until session is dismissed/committed,)
+            // and we'd check back to see if it's completed whenever we have a chance to update the completion list, i.e. when user typed another
+            // character, a filter was selected, etc. If so, those items will be added as part of the refresh.
```

Comment:
It's not possible with current API. Also I'm not sure live update is desirable here, unless we can figure out a way to do so without changing what's already shown to the user. Because unlike async lighbulb, the addition of new items might drastically change the state of the list (i.e. sorting and selection)